File: fixtures/scale/typescript/large/src/checks/CheckParitymix.ts
import { Paritymix } from '../ciphers/Paritymix';
import { ParitymixSpec } from '../specs/ParitymixSpec';
import { CorpusMask } from '../corpus/CorpusMask';

export class CheckParitymix {
  static run(): boolean {
    const spec = ParitymixSpec.get();
    for (const text of CorpusMask.texts()) {
      const sealed = Paritymix.encrypt(text, spec.key);
      const opened = Paritymix.decrypt(sealed, spec.key);
      if (opened !== text) return false;
    }
    return spec.category === 'mask';
  }
}

File: fixtures/scale/lua/large/checks/check_paritymix.lua
local paritymix = require("ciphers.paritymix")
local paritymix_spec = require("specs.paritymix_spec")
local corpus_mask = require("corpus.corpus_mask")

local function check_paritymix()
  local spec = paritymix_spec.get()
  for _, text in ipairs(corpus_mask.corpus_mask()) do
    local sealed = paritymix.paritymix_encrypt(text, spec.key)
    local opened = paritymix.paritymix_decrypt(sealed, spec.key)
    if opened ~= text then return false end
  end
  return spec.category == "mask"
end

return { check_paritymix = check_paritymix }
